library/src/main/java/org/xmtp/android/library/Client.kt

```diff
@@ -90,9 +90,10 @@ class Client() {
     var logger: XMTPLogger = XMTPLogger()
     val libXMTPVersion: String = getVersionInfo()
     var installationId: String = ""
-    private var v3Client: FfiXmtpClient? = null
+    var v3Client: FfiXmtpClient? = null
     var dbPath: String = ""
     lateinit var inboxId: String
+    var hasV2Client: Boolean = false
```

Comment:
Just a nit - looks like it is currently impossible to create a `Client` without `hasV2Client` being `true`, but it is pretty hard to tell that from this code. (you need to visually verify that all functions that return Client either call `buildFromV1Bundle` or `loadOrCreateKeys`)

Since currently all code paths set this to `true`, I wonder if we should just set it to true by default, instead of false, and add the hasV2Client = `false` later on when there is a path that actually makes this possible.

Reply:
Hmmm you know this is a good point. I actually think it's going to be really hard to validate the correctness of all our work without having the ability to make a pure v3 client. I'm going to open a separate PR today that so that we can make certain this stuff is working like expected in V3 and not just a side effect of the dual sending.

Reply:
From creating this PR I align with you it should be defaulted to true https://github.com/xmtp/xmtp-android/pull/298/files